utils-intervalrdd/src/main/scala/org/bdgenomics/utils/interval/array/IntervalArray.scala

```diff
@@ -108,6 +108,13 @@ object IntervalArray extends Serializable {
 trait IntervalArray[K <: Interval[K], T] extends Serializable {
   val array: Array[(K, T)]
   val maxIntervalWidth: Long
+  // maintains the last index returned from the most recent binarySearch result.
+  // !!!
+  // WARNING: NOT THREAD SAFE
+  // TO SAFELY USE THIS VARIABLE IN A SPARK CONTEXT, CREATE A SHALLOW COPY OF
+  // THE INTERVAL ARRAY.
+  // !!!
+  private var optLastIndex: Option[Int] = None
```

Comment:
This is strictly used in get to cache the previous interval so in the case that the next interval also overlaps it, we can skip the binary search. More docs to come about this.